training/run_distillation.py

```diff
@@ -976,6 +999,13 @@ def set_trainable_parameters(module, requires_grad=False):
     if training_args.freeze_encoder:
         set_trainable_parameters(student_model.model.encoder, requires_grad=False)
         student_model.model.encoder.gradient_checkpointing = False
+    
+    if training_args.freeze_decoder:
+        set_trainable_parameters(student_model.model.decoder, requires_grad=False)
+        student_model.model.decoder.gradient_checkpointing = False
+        # un-freeze LM head parameters (and consequently word embeddings), frozen when frozing decoder since tied word embedding and LM head
+        for p in student_model.proj_out.parameters():
+            p.requires_grad = True 
```

Comment:
```suggestion
        # un-freeze LM head parameters (and consequently word embeddings), frozen when freezing decoder since tied word embedding and LM head
        set_trainable_parameters(student_model.model.encoder, requires_grad=True)
```

Reply:
If think you meant  `set_trainable_parameters(student_model.proj_out, requires_grad=True)`. Implemented in [7280623](https://github.com/huggingface/distil-whisper/pull/126/commits/7280623e6bfd8343e455fc465046c3fd64b4f8fd)